### PR TITLE
Fix return for virtual product

### LINKED ISSUE #16275: Prevent customers from returning a full virtual order
**Is your feature request related to a problem? Please describe:**

When an order contains only virtual products, the Merchandise Returns tab displays the following message: Merchandise returns are not applicable for virtual orders. So we let the merchant know virtual products (and by extension, fully virtual orders) cannot be returned.

But once a virtual order is noted as delivered, the customer can connect to his/her account, go in the details of his/her virtual order, select all virtual products and decide to return them.

Back from a merchant's side, the Merchandise Returns tab remains the same, with the message displayed and the customer return not listed.

**Describe the solution you'd like:**

It would make sense to prevent the customer from returning his/her virtual order. Perhaps by displaying the same message?

cf. https://github.com/PrestaShop/PrestaShop/pull/16084#discussion_r338060225

### PR DESCRIPTION
**Pull request by ghost**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | Customer can return virtual product this PR allow to return only no virtual product.
| Type?             | improvement
| Category?         | FO
| BC breaks?        | no
| Deprecations?     | no
| Fixed ticket?     | Fixes https://github.com/PrestaShop/PrestaShop/issues/16275
| How to test?      |  2 steps : Made an order with only virtual product and check return Made an order with a virtual product and a no virtual product check the return. Order need to be shipped and return enable in the back office
| Possible impacts? | No

Can you confirm that I can delete {if $order.details.is_returnable} because I cannot find is_returnable anywhere ?

Order with one virtual product and one no virtual :

Capture d’écran de 2021-07-01 08-47-39

Order with only virtual product :

Capture d’écran de 2021-07-01 08-48-01

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/25165)

### CODE REVIEW

**Comment by ghost** [comment]
Wip after check we do not have all the articles in the summary  for the second scenario

File: controllers/front/OrderDetailController.php [hunk 1]
```diff
@@ -189,6 +189,7 @@ public function initContent()
 
                 $this->context->smarty->assign([
                     'order' => $this->order_to_display,
+                    'isVirtual' => $order->isVirtual(),
```
Comment (jf-viguier) [inline controllers/front/OrderDetailController.php:192]:
```suggestion
                    'orderIsVirtual' => $order->isVirtual(),
```

File: themes/classic/templates/customer/order-detail.tpl [hunk 2]
```diff
@@ -148,7 +148,7 @@
   {$HOOK_DISPLAYORDERDETAIL nofilter}
 
   {block name='order_detail'}
-    {if $order.details.is_returnable}
+    {if $order.details.is_returnable && $isVirtual == false}
```
Comment (jf-viguier) [inline themes/classic/templates/customer/order-detail.tpl:151]:
```suggestion
    {if $order.details.is_returnable && $orderIsVirtual == false}
```

**Comment by ghost** [comment]
Thank you @jf-viguier

**Comment by Progi1984** [comment]
@okom3pom `isReturnable` doesn't exist in `src/Adapter/Presenter/Order/OrderDetailLazyArray.php` but a `getIsReturnable` method exists.

**Comment by ghost** [comment]
Sorry I close this one i am a noob with rebase

**Pull request closed without merging**